### PR TITLE
Milestone 3

### PR DESCRIPTION
**Pull request by Lidya1234**
-Implement two-way friendship by adding two rows for a friendship
-Fixed the update method of the friendship model
-Added the current username to the navigation bar.
--IN THE DATABASE.YML FILE CHANGE THE USERNAME AND PASSWORD TO YOURS.

### CODE REVIEW

**Review by PhillipUg — CHANGES_REQUESTED** [review]
Hey team, 👋

You have done an amazing job implementing the project requirements of this project.

However, there are still some changes 👇 you need to implement to approve this request 🔍

### Changes Required ♻️ 

- [x] I am unable to get any of the `add_friend` functionality to work in your app. Fix it and resubmit for review.

#### Please, make REQUIRED changes and submit your request again! 😀

If you have any questions, feel free to send me a message through Slack as **Phillip Musiime**
I would be happy to help! 👩‍💻 

Happy coding 🤓

File: db/migrate/20201209161237_add_column_rerails.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class AddColumnRerails < ActiveRecord::Migration[5.2]
+  def change
+  end
+end
```
Comment (tirthajyoti-ghosh) [inline db/migrate/20201209161237_add_column_rerails.rb:4]: Please remove this migration as it's empty.

File: db/migrate/20201210083801_add_column_to_friendships.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddColumnToFriendships < ActiveRecord::Migration[5.2]
+  def change
+    add_column :friendships, :requester_id
```
Comment (tirthajyoti-ghosh) [inline db/migrate/20201210083801_add_column_to_friendships.rb:3]: I got an error while running this migration 👇 .

image

You forgot to mention the data type. Please rectify this.

**Review by tirthajyoti-ghosh — CHANGES_REQUESTED** [review]
# Changes required 🛑 

## Impressive progress so far 💪! Thanks for making all the changes 👏.

Great job implementing the V2 friendship. It works perfectly!

### However, we need to work on some changes before we can merge.

- I see that you have opened another PR for this milestone. Please don't do this. Opening new PRs for every requested change will confuse the reviewers and you.

- I wasn't able to run `rails db:migrate` after setting it up locally. I got an error. I have left a comment below about this.

That's it!

Your project looks absolutely fantastic. Just make these changes and apply for another code review.

Happy coding 👨‍💻 

*Need help? Shoot me a message on Slack - [@Ghosh](https://app.slack.com/client/T47CT8XPG/DRDM6LAQN/user_profile/URB1MHXK6)*
<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

**Review by Spayco — CHANGES_REQUESTED** [review]
# Changes Requested 

Hello Team! 👋 
Awesome job so far!! 👍 you still need to make some changes though! 

## Required Changes ♻️ 

- [ ] I'm still getting an error when trying to migrate, [here's the log](https://pastebin.com/AQK3Z3Hn) I get, it should help you find the issue!

### please make the appropriate changes and ask for another code review ✔️ 

Happy Coding!

Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

**Comment by PhillipUg** [comment]
Hey team, this is not a review. Its just a piece of unsolicited advice.

Go to `config/database.yml` and remove `username`, `password`, `host` and `port` fields. They are the reason TSEs are having problems running migration or setting  up your database. I have reviewed this project twice and i mentioned this in my first review on the second milestone. I always solve it by commenting out those fields in `database.yml`. 

Otherwise happy coding 😎

**Review by elmejdki — APPROVED** [review]
# status: approved :heavy_check_mark: 

Great job @Lidya1234 :clap: :clap: :clap:. Thanks for making the required changes by the previous TSE's :100: :+1:.
 
now you can merge your project is approved :smiley: 

## Happy Coding :tophat:

**Pull request merged**